### PR TITLE
Fix issue for storage and key vault

### PR DESCRIPTION
**Pull request by felipmiguel**
When integrating in a vnet:
- Fix issue for redis and azure spring cloud. There were overlapping
address spaces. #226 
- KeyVault and Azure Storage failed because both resources were not reachable from the machine executing the script. #225

### CODE REVIEW

**Review by cmaneu — CHANGES_REQUESTED** [review]

File: src/main/resources/nubesgen/terraform/main.tf.mustache [hunk 1]
```diff
@@ -22,6 +22,16 @@ locals {
   // If an environment is set up (dev, test, prod...), it is used in the application name
   environment = var.environment == "" ? "dev" : var.environment
 }
+{{#AllowingClientIpRequired}}
+
+data "http" "myip" {
+  url = "http://ipv4.icanhazip.com"
```
[the remark this answers is not in the document]
Comment (felipmiguel) [reply src/main/resources/nubesgen/terraform/main.tf.mustache:28]: If no internal service available maybe it is possible to expose an endpoint at nubesgen.com that returns the client IP. something like:
https://nubesgen.com/myip

[the remark this answers is not in the document]
Comment (cmaneu) [reply src/main/resources/nubesgen/terraform/main.tf.mustache:28]: @felipmiguel  After some digging, it seems we (Microsoft) can use http://whatismyip.akamai.com/.
Also there is http://whatismyip.akamai.com/advanced for more details (not useful in this context, but useful to know)

[the remark this answers is not in the document]
Comment (felipmiguel) [reply src/main/resources/nubesgen/terraform/main.tf.mustache]: Maybe it is possible to use deployment scripts in bicep to get the IP. https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/deployment-script-bicep#sample-bicep-files

[the remark this answers is not in the document]
Comment (cmaneu) [reply src/main/resources/nubesgen/terraform/main.tf.mustache]: I doubt this will work. Deployment scripts are run on a remote container, not on the deployment agent itself, so we'll not get the right IP address :/

[the remark this answers is not in the document]
Comment (felipmiguel) [reply src/main/resources/nubesgen/terraform/main.tf.mustache]: In this case for bicep I only see executing a tool out of the deployment script and passing the IP as a parameter.

File: src/main/resources/nubesgen/bicep/main.bicep.mustache [hunk 2]
```diff
@@ -317,3 +317,18 @@ module function 'modules/function/function.bicep' = {
   }
 }
 {{/applicationTypeFunction}}
```
Comment (cmaneu) [inline src/main/resources/nubesgen/bicep/main.bicep.mustache:319]: This code should belong to another PR

File: src/main/resources/nubesgen/bicep/modules/spring-cloud/spring-cloud.bicep.mustache [hunk 3]
```diff
@@ -0,0 +1,78 @@
+// Azure Spring Cloud - Bicep module
```
Comment (cmaneu) [inline src/main/resources/nubesgen/bicep/modules/spring-cloud/spring-cloud.bicep.mustache:1]: This code should belong to another PR

**Review by cmaneu — CHANGES_REQUESTED** [review]
The code adding some features to Bicep should be moved to another PR

File: src/main/resources/nubesgen/bicep/main.bicep.mustache (hunk 2 above)
Comment (felipmiguel) [reply src/main/resources/nubesgen/bicep/main.bicep.mustache]: removed from this pr

File: src/main/resources/nubesgen/bicep/modules/spring-cloud/spring-cloud.bicep.mustache (hunk 3 above)
Comment (felipmiguel) [reply src/main/resources/nubesgen/bicep/modules/spring-cloud/spring-cloud.bicep.mustache]: removed from this pr

**Review by cmaneu — APPROVED** [review]

**Comment by cmaneu** [comment]
@felipmiguel while deploying, I've noted that the Key Vault is open for all networks, so I'm not sure it's ready for merge.
image

**Comment by cmaneu** [comment]
What is weird is the network ACL is well generated
```
  network_acls {
    default_action             = "Allow"
    bypass                     = "None"
    virtual_network_subnet_ids = [var.subnet_id]
    ip_rules                   = [var.myip]
  }
  ```

**Comment by cmaneu** [comment]
Closed, replaced by #228
